Add Real Location Data for Event

allysmith12 opened the pull request:
This adds the front end and backend functionality for a user to input a location.

This utilizes Google Maps to show a location picker that autocomplete the user's input.
This code also creates a new knex migration to add the appropriate table columns and edits the addEvent route to include the location data. 

Current API of create event:
<img width="390" alt="Screen Shot 2020-10-30 at 1 16 57 PM" src="https://user-images.githubusercontent.com/70663801/97736522-31fb8480-1ab2-11eb-9144-233c3d6af3cf.png">
<img width="407" alt="Screen Shot 2020-10-30 at 1 17 23 PM" src="https://user-images.githubusercontent.com/70663801/97736554-40e23700-1ab2-11eb-9068-24a931e0204e.png">

<img width="404" alt="Screen Shot 2020-10-30 at 1 17 43 PM" src="https://user-images.githubusercontent.com/70663801/97736576-4ccdf900-1ab2-11eb-8775-1ee485284890.png">

EDIT -- this PR cannot be merged until the secret environment variable for the google maps api is set up for production

allysmith12 committed:
Follow up to fix user id setting

allysmith12 committed:
Follow up to fix user id setting

allysmith12 committed:
Add ScrollView

allysmith12 committed:
Move getting the userId token from the frontend to the backend

allysmith12 committed:
Move getting the userId token from the frontend to the backend

allysmith12 committed:
Created React Navigation to move between Schedule and Create event screen.

Added button to go to CreateEvent screen. Submit on createEvent screen will navigate user back to Schedule screen.

allysmith12 committed:
Created React Navigation to move between Schedule and Create event screen.

Added button to go to CreateEvent screen. Submit on createEvent screen will navigate user back to Schedule screen.

allysmith12 committed:
Merge branch 'main' into create-event-follow-up

allysmith12 committed:
Move ScheduleHomePage out of ScheduleNavigator and into Schedule module

allysmith12 committed:
Move ScheduleHomePage out of ScheduleNavigator and into Schedule module

allysmith12 committed:
Remove unused imports

allysmith12 committed:
Remove unused imports

allysmith12 committed:
Merge branch 'create-event-follow-up' into event-location

allysmith12 committed:
Merge branch 'main' into event-location

zusoomro commented on mobile/src/events/LocationPicker.tsx:
```diff
@@ -0,0 +1,60 @@
+import { GooglePlacesAutocomplete } from "react-native-google-places-autocomplete";
+
+// // FIGURE OUT HOW TO RESTRICT THIS // encode it 
+const api_key: string = "AIzaSyAYRQIS7VyzXVVlrcsch2SQO5qtW3dGOV4";
```
We should try to keep api keys out of github - look at using environment variables to keep this stored in local files. So google smth like "How to use environment variables api key" or something like that.

zusoomro commented on mobile/src/events/LocationPicker.tsx:
```diff
@@ -0,0 +1,60 @@
+    const [formattedAddress] = useField(props.formattedAddress);
+
+    const locationPicked = (data, details) => {
+        console.log("Selected Location")
```
Do we need this console.log?

allysmith12 replied:
Done

zusoomro reviewed (commented):
Looks good! Check out my comments below and I think you also need to format some things before we should merge.

allysmith12 committed:
Added Location Picker to Front End.

User can choose location via LocationPicker and the value of the location is placed inside Formik.

Note: this does not work bc I removed the api key

allysmith12 committed:
Added Location Picker to Front End.

User can choose location via LocationPicker and the value of the location is placed inside Formik.

Note: this does not work bc I removed the api key

allysmith12 committed:
Add location data to database in addEvent request.

allysmith12 committed:
Add location data to database in addEvent request.

allysmith12 committed:
Remove unused text box for address

allysmith12 committed:
Remove unused text box for address

allysmith12 committed:
Adding back in api_key imports -

- confer with Zulfi how he wants this to work

allysmith12 committed:
Adding back in api_key imports -

- confer with Zulfi how he wants this to work

allysmith12 committed:
Add environement var for maps key

allysmith12 committed:
Add environement var for maps key

allysmith12 committed:
Remove console.log

allysmith12 committed:
Remove console.log

allysmith12 committed:
Yarn lock

allysmith12 committed:
package.json

allysmith12 committed:
delete yarn 2.lock

allysmith12 committed:
Merge branch 'event-location' of https://github.com/zusoomro/wigo into event-location

allysmith12 committed:
Formatting

allysmith12 committed:
Fix yarn.lock to re-add neccessary dependenices + fix issues from rebase

allysmith12 committed:
Fix mistake from rebase

zusoomro commented on api/src/routes/events.ts:
```diff
@@ -4,15 +4,31 @@ import auth, { AuthRequest } from "../authMiddleware";
     ownerId: id,
     name,
-    address,
+    formattedAddress: formattedAddress,
```
You can just do `formattedAddress,` here, since they're called the same thing, I think.

allysmith12 replied:
Donee!

zusoomro commented on mobile/src/events/DatePicker.tsx:
```diff
@@ -4,7 +4,7 @@ import { View, Text, Platform, StyleSheet } from "react-native";
 import DateTimePicker from "@react-native-community/datetimepicker";
 
-const DatePicker: React.FC<{}> = ({ ...props }) => {
+const DatePicker: React.FC<{ name: string; date: Date }> = (props) => {
```
Love to see it

allysmith12 replied:
Let me know when I can merge this bad boy!

allysmith12 committed:
Formatting fixed

allysmith12 committed:
Merge branch 'main' into event-location

allysmith12 committed:
follow up from merge

allysmith12 committed:
formatting

allysmith12 committed:
Removed bad route for get event.
